attributes/geo:proj/README.md

```diff
@@ -0,0 +1,237 @@
+
+### Spatial Dimension Identification
+
+In this extension, "spatial dimensions" refers to the dimension names of 2D/3D arrays within this group to which the projection definition applies. This extension is designed for regular grids where dimensions directly correspond to spatial axes.
+
+The extension identifies these array dimensions through:
+
+1. **Explicit Declaration** (recommended): Use `spatial_dimensions` to specify dimension names
+2. **Pattern-Based Detection** (fallback): Automatically detect spatial dimensions using patterns defined by this extension
+
+#### Explicit Declaration
+
+```json
+{
+  "geo:proj": {
+    "spatial_dimensions": ["latitude", "longitude"]
+  }
+}
+```
+
+#### Pattern-Based Detection
+
+If `spatial_dimensions` is not provided, implementations should scan `dimension_names` for these patterns defined by this extension (in order):
+
+- ["y", "x"] or ["Y", "X"]
+- ["lat", "lon"] or ["latitude", "longitude"]
+- ["northing", "easting"]
+- ["row", "col"] or ["line", "sample"]
+
+The first matching pair determines the spatial dimensions. **Important**: When dimensions like "X" and "Y" are found, they are always interpreted as [Y, X] (following lat/lon convention), regardless of their actual order in the Zarr array's `dimension_names`.
+
+### Validation Rules
+
+- **Shape Inference**: Once spatial dimensions are identified (either explicitly through `spatial_dimensions` or through pattern-based detection), their sizes are obtained from the Zarr array's shape metadata
+- **Spatial Dimension Order**: The spatial dimension order is always [y/lat/northing, x/lon/easting]
+- **Error Handling**: If spatial dimensions cannot be identified through either method, implementations MUST raise an error
+- **Semantic Identity Requirement**: If more than one CRS representation (`code`, `wkt2`, `projjson`) is provided, they MUST be semantically identical (i.e., describe the same coordinate reference system). Implementations SHOULD validate this consistency and raise an error if the representations describe different CRS
+
+### Shape Reconciliation
+
+The shape of spatial dimensions is determined by:
+
+1. Identifying the spatial dimensions using either `spatial_dimensions` or pattern-based detection
+2. Looking up these dimension names in the Zarr array's `dimension_names`
+3. Using the corresponding sizes from the array's `shape` attribute
+
+This approach avoids redundancy and ensures consistency by using the array's own metadata rather than duplicating shape information.
+
+## Examples
+
+### Example 1: Simple Web Mercator Raster (Group Level)
+
+```json
+{
+  "zarr_format": 3,
+  "node_type": "group",
+  "attributes": {
+    "geo:proj": {
+      "code": "EPSG:3857",
+      "transform": [156543.03392804097, 0.0, -20037508.342789244, 0.0, -156543.03392804097, 20037508.342789244],
+      "bbox": [-20037508.342789244, -20037508.342789244, 20037508.342789244, 20037508.342789244]
```

Comment:
One point of confusion for me in STAC is whether the bbox refers to the outer bounds of the cells or the coordinate values for raster data? From this, it looks like it's the coordinate values which will always be narrower than the outer bounds of the cells for raster data. Should we specify that in this document?